### ISSUE TITLE
Question about SIGNIFICANT_DIGITS precision

### ISSUE THREAD

**Issue by jackpotcityco** [ccxt/ccxt#8574]
OS: Windows Server 2019 Datacenter
Programming Language version: Node.js 15.3.0
CCXT version: 1.38.89
Exchange: All Exchanges
Method: SIGNIFICANT_DIGITS 

Hello!

I have red about precision and limits here:
https://github.com/ccxt/ccxt/wiki/Manual#precision-and-limits

**I wonder about the below that is written in the manual:**
_SIGNIFICANT_DIGITS – counts non-zero digits only, some exchanges (bitfinex and maybe a few other) implement this mode of counting decimals. With this mode of precision, the numbers in market['precision'] designate the Nth place of the last significant (non-zero) decimal digit after the dot._

If the precision must be 5, I then wonder for the below examples. I think I got example 1 correct. But what is correct for example 2 and 3. I am not 100% sure I understand this?
**Example1: 12.000123456**
This must be: 12.00012345 ?

**Example2: 13.0101234** 
How should this be rounded down to 5 digits?

**Example3: 14.010120346** 
How should this be rounded down to 5 digits?

-------------------------------------------------------------

I have tested this code below and it returns: 55000 for the roundedNum. I wonder if that is correct. 
Shouln't it be: 55000.12346 which is **5 precision.price** decimals after the dot?
```
'use strict';
const ccxt = require('ccxt');

(async () => {
    let exchange = new ccxt.bitfinex(); 
    var markets = await exchange.load_markets();
    var market = markets['BTC/USD']; 
    var precision = market['precision']; //here are precision values for price and amount
    var countingmode = exchange.precisionMode; //0,1,2 (DECIMAL_PLACES, SIGNIFICANT_DIGITS, TICK_SIZE)
    ///
    //Rounding mode is always TRUNCATE for amounts, ROUND for prices.
    //(x, roundingMode, numPrecisionDigits, countingMode, paddingMode = NO_PADDING)

    var roundedNum = exchange.decimalToPrecision(55000.123456789, ccxt.ROUND, precision.price, countingmode); 
    console.log(roundedNum + "___" + precision.price + "___" + countingmode);
})()
```

Thank you!

**Comment by kroitor** [comment]
@jackpotcityco significant digits = the number of digits to the right of the first non-zero digit in your number.

```JavaScript
//      ↓ 5th digit after the first non-zero digit
//      ↓
   12.000123456 = 12.000 = 12.0 = 12
// ↑ 
// ↑ first non-zero digit
```

Cutting off everything after the 5th digits you're left with `12.000` and after truncating zeroes that yields `12.0` or `12`

```JavaScript
//      ↓ 5th digit after the first non-zero digit
//      ↓
   13.0101234 = 13.010 = 13.01
// ↑ 
// ↑ first non-zero digit
```

Example 3 is the same as Example 2.

Also, check this:

```JavaScript
//          ↓ 5th digit after the first non-zero digit
//          ↓
   0.000101234 = 0.00010123
//      ↑ 
//      ↑ first non-zero digit
```

All questions about precision are answered in this self-explaining test: 

- https://github.com/ccxt/ccxt/blob/master/js/test/base/functions/test.number.js

Let us know if that does not help.

**Comment by jackpotcityco** [comment]
I am sorry for a late answer, but that helped alot. The link to the test.number.js was great!
I can follow all logic there perfectly!

Thank you for help!

**Comment by jackpotcityco** [comment]
I have gone through the test.number.js and I have followed all examples there.

I think I have found one scenario which is the below that I couldn't find. 
This when using ´ccxt.Round´ and ´-1´ and ´SIGNIFICANT_DIGITS´.

If I use any of: ´-1´, ´-2´ or ´-3´, they all produce the result: ´100´

I don't know if the below is a valid scenario? If it is a valid scenario how many SIGNIFICANT_DIGITS will ´-1´ return.
It seems something is wrong here?

``
let exchange = new ccxt.bitfinex(); 
var roundedNum = exchange.decimalToPrecision(51.57, ccxt.ROUND, -1, 1);
console.log(roundedNum);
``
